pandas/io/formats/csvs.py

```diff
@@ -33,27 +35,26 @@ class CSVFormatter:
     def __init__(
         self,
         obj,
-        path_or_buf=None,
-        sep=",",
-        na_rep="",
+        path_or_buf: Optional[FilePathOrBuffer[str]] = None,
+        sep: str = ",",
+        na_rep: str = "",
         float_format=None,
         cols=None,
-        header=True,
-        index=True,
+        header: Union[bool, Sequence[str]] = True,
+        index: bool = True,
         index_label=None,
-        mode="w",
-        encoding=None,
+        mode: str = "w",
+        encoding: Optional[str] = None,
         compression="infer",
-        quoting=None,
-        line_terminator="\n",
-        chunksize=None,
-        quotechar='"',
-        date_format=None,
-        doublequote=True,
+        quoting: Optional[int] = None,
+        line_terminator: Optional[str] = "\n",
```

Comment:
```suggestion
        line_terminator: str = "\n",
```

Don't think optional

Reply:
Should I keep it as

```python
line_terminator: Optional[str] = "\n"
```

?

---

mypy's log:

```
$ mypy pandas
pandas/core/generic.py:3164: error: Argument "line_terminator" to "CSVFormatter" has incompatible type "Optional[str]"; expected "str"
Found 1 error in 1 file (checked 816 source files)
```

Reply:
Hmm there is a minor disconnect between `to_csv` and the `CSVFormatter` class on this. Can you create a follow up issue? Not sure of easiest way to correct